llvm/test/CodeGen/X86/any_extend_vector_inreg_of_broadcast.ll

```diff
@@ -750,7 +750,7 @@ define void @vec128_i16_widen_to_i32_factor2_broadcast_to_v4i32_factor4(ptr %in.
 ; AVX512BW-SLOW-LABEL: vec128_i16_widen_to_i32_factor2_broadcast_to_v4i32_factor4:
 ; AVX512BW-SLOW:       # %bb.0:
 ; AVX512BW-SLOW-NEXT:    vmovdqa64 (%rdi), %zmm0
-; AVX512BW-SLOW-NEXT:    vmovdqa {{.*#+}} xmm1 = [0,9,0,11,0,13,0,15]
+; AVX512BW-SLOW-NEXT:    vpmovsxbw {{.*#+}} xmm1 = [0,9,0,11,0,13,0,15]
```

Comment:
Can small memory size beat defects in micro arch, e.g.

- vpmovsxbw uses one more port
- vmovdqa can be eliminated by uarch

Reply:
That was my reasoning for always preferring vzload/broadcast over vextload for the same constant load size

Reply:
I don't get the point. Doesn't vzload have the same TP/ports with vextload?
And if vzload is preferred, why not a `vpmovzxbw` here?
My initial question was about "load size" vs. "inst cost", but we already using broadcast which has the cost too (e.g., vpbroadcastb/w). So it should not be a problem.

Reply:
This is how I see it regarding preference between vzload/broadcast/vextload:
1. vzload shouldn't ever need a shuffle port to zero the upper elements and the fp/int domain versions are equally available so we don't introduce a domain crossing penalty.
2. broadcast sometimes need a shuffle port (especially for 8/16-bit variants), AVX1 only has fp domain broadcasts but AVX2+ have good fp/int domain equivalents
3. vextload always needs a shuffle port and is only ever int domain

Reply:
Yes, a comment is great.

> vzload shouldn't ever need a shuffle port to zero the upper elements

But this doesn't match what I got from uops.info, where the `vpmovzx` has the same shuffle port as `vpmovsx`

Reply:
Do you mean vpmovzx (vzextload which I haven't added yet due to some shuffle comment problems) or vmovd/vmovq (vzload) which just zeros the upper elements?

Reply:
I've added a comment describing the preferences